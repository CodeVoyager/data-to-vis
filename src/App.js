import React, { Component } from 'react';
import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';
import {connect} from 'react-redux'
import {
    availableCurrencies,
    currency,
    data,
    startDate,
    endDate,
    isLoading,
} from './selectors';
import {
    setAvailableCurrencies as setAvailableCurrenciesAction,
    setCurrency as setCurrencyAction,
    setEndDate as setEndDateAction,
    setStartDate as setStartDateAction,
    setData as setDataAction
} from './actions';
import store from './store'
import api from './api'
import {
    DATE_FORMAT,
    PREFERED_CURRENCY
} from './config'
import Loader from './Loader';
import Chart from './Chart';
import moment from 'moment';
const R = require('ramda');

const mapStateToProps = state => {
    return {
        availableCurrencies: availableCurrencies(state),
        currency: currency(state),
        data: data(state),
        startDate: startDate(state),
        endDate: endDate(state),
        isLoading: isLoading(state),
    };
}
class App extends Component {
    onStartDateChange (date) {
        store.dispatch(setStartDateAction(date));
    }
    onEndDateChange (date) {
        store.dispatch(setEndDateAction(date));
    }
    onCurrencyChange (event) {
        store.dispatch(setCurrencyAction(event.target.value));
    }
    onCurrenciesAvailable (currencies) {
        let currency = R.find(R.equals(PREFERED_CURRENCY), currencies);

        store.dispatch(setCurrencyAction(currency || currencies[0]));
        store.dispatch(setAvailableCurrenciesAction(currencies));
    }
    componentDidMount () {
        api.getCurrencies(this.onCurrenciesAvailable);
        if (R.isNil(this.props.startDate)) {
            store.dispatch(setStartDateAction(moment().subtract(1, 'years')));
        }
        if (R.isNil(this.props.endDate)) {
            store.dispatch(setEndDateAction(moment()));
        }
    }
    onSubmit () {
        api.getData(this.props.startDate, this.props.endDate, this.props.currency, (data) => {
            store.dispatch(setDataAction(data));
        });
    }
    render() {
        let availableCurrenciesOptions = null;
        let margin = {top: 20, right: 20, bottom: 30, left: 50};

        if (this.props.availableCurrencies && this.props.availableCurrencies.length) {
            availableCurrenciesOptions = this.props.availableCurrencies.map(currency => {
                return <option key={currency} value={currency}>{currency}</option>;
            });
        }

        return (
            <div className="app">
                <header className="header">
                    <h1 className="title">Data to Vis</h1>
                    <h2 className="sub-title">Bitcoin price index chart</h2>
                </header>
                <div className="steps">
                    <div className="step">
                        <div className="header">
                            1. Select dates
                        </div>
                        <div className="cols">
                            <div className="col-half">
                                    <DatePicker
                                        className="input"
                                        selected={this.props.startDate}
                                        onChange={this.onStartDateChange}
                                        maxDate={this.props.endDate}
                                        dateFormat={DATE_FORMAT}
                                        placeholderText="Click to select start date"
                                    />
                                </div>
                                <div className="col-half">
                                    <DatePicker
                                        className="input"
                                        selected={this.props.endDate}
                                        onChange={this.onEndDateChange}
                                        minDate={this.props.startDate}
                                        dateFormat={DATE_FORMAT}
                                        placeholderText="Click to select end date"
                                    />
                            </div>
                        </div>
                    </div>
                    <div className="step">
                        <div className="header">
                            2. Select currency
                        </div>
                        <div className="cols">
                            <div className="col-full">
                                {
                                    (this.props.availableCurrencies && this.props.availableCurrencies.length)
                                    ? (
                                        <select className="input" value={this.props.currency} onChange={this.onCurrencyChange}>
                                            {availableCurrenciesOptions}
                                        </select>
                                    ) : null
                                }

                            </div>
                        </div>
                    </div>
                    <div className="step">
                        <div className="header">
                            3. Click submit
                            <div className="cols">
                                <div className="col-full">
                                    <button disabled={!(this.props.availableCurrencies && this.props.availableCurrencies.length)} onClick={this.onSubmit.bind(this)} className="input submit">
                                        Submit
                                    </button>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
                {
                    (this.props.data && this.props.data.length) ? (
                        <Chart
                            data={this.props.data}
                            width={860}
                            height={400}
                            margin={margin}
                            xKey={0}
                            yKey={1}
                        />
                    ) : null
                }
                {
                    this.props.isLoading ? (
                        <Loader/>
                    ) : null
                }
            </div>
        );
    }
}

export default connect(mapStateToProps)(App);
export {App};