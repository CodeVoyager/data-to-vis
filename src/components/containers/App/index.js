import React, {Component} from 'react'
import {
    Route,
    BrowserRouter as Router
} from 'react-router-dom';
import { Provider } from 'react-redux'
import store from '../../../store';
import BPI from '../BPI';
import Grid from '../../presentational/Grid';

class App extends Component {
    render () {
        return (
            <Provider store={store}>
                <Router>
                    <div className="app">
                        <Route exact path="/" component={Grid}/>
                        <Route path="/bpi" component={BPI}/>
                    </div>
                </Router>
            </Provider>
        );
    }
}

export default App;
