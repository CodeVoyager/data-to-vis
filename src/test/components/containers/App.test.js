import React from 'react';
import ReactDOM from 'react-dom';
import {App} from '../../../components/containers/App';
import renderer from 'react-test-renderer';
import moment from 'moment';


describe('App component', function () {
    it('renders without crashing', function () {
        const div = document.createElement('div');
        ReactDOM.render(<App />, div);
    });

    it('renders correctly', function () {
        const startDate = moment('20171031', 'YYYYMMDD');
        const endDate = startDate.subtract(30, 'days');
        const currencies = ['USD', 'PLN'];
        const currency = 'USD';
        const data = [
            ['2013-09-01', 128.2597],
            ['2013-09-02', 127.3648]
        ];

        const tree1 = renderer.create(<App />).toJSON();
        expect(tree1).toMatchSnapshot();

        const tree2 = renderer.create(<App startDate={startDate} />).toJSON();
        expect(tree2).toMatchSnapshot();

        const tree3 = renderer.create(<App endDate={endDate} />).toJSON();
        expect(tree3).toMatchSnapshot();

        const tree4 = renderer.create(<App availableCurrencies={currencies} />).toJSON();
        expect(tree4).toMatchSnapshot();

        const tree5 = renderer.create(<App currency={currency} />).toJSON();
        expect(tree5).toMatchSnapshot();

        const tree6 = renderer.create(<App data={data} />).toJSON();
        expect(tree6).toMatchSnapshot();

        const tree7 = renderer.create(<App isLoading={false} />).toJSON();
        expect(tree7).toMatchSnapshot();

        const tree8 = renderer.create(<App isLoading={true} />).toJSON();
        expect(tree8).toMatchSnapshot();

        const tree9 = renderer.create(<App data={data} isLoading={true} currency={currency} availableCurrencies={currencies} startDate={startDate} endDate={endDate} />).toJSON();
        expect(tree9).toMatchSnapshot();

        const tree10 = renderer.create(<App data={data} isLoading={false} currency={currency} availableCurrencies={currencies} startDate={startDate} endDate={endDate} />).toJSON();
        expect(tree10).toMatchSnapshot();
    });

    it('should have onCurrencyChange method', function () {
        expect(App.prototype.onCurrencyChange).not.toBeUndefined();
    });

    it('should onCurrencyChange pass valid data to props.onCurrencyChange on invoke', function (done) {
        const fn = jest.fn();
        const component = renderer.create(<App onCurrencyChange={fn} />);
        const VALUE = 'VALUE';

        component.getInstance().onCurrencyChange({
            target: {
                value: VALUE
            }
        });

        expect(fn.mock.calls.length).toBe(1);
        expect(fn.mock.calls[0][0]).toBe(VALUE);

        done();
    });

    it('should have onCurrenciesAvailable method', function () {
        expect(App.prototype.onCurrenciesAvailable).not.toBeUndefined();
    });

    it('should onCurrenciesAvailable pass valid data to props.onCurrenciesAvailable on invoke', function (done) {
        const fn = jest.fn();
        const component = renderer.create(<App onCurrenciesAvailable={fn} />);
        const DATA = 'DATA';

        component.getInstance().onCurrenciesAvailable(DATA);

        expect(fn.mock.calls.length).toBe(1);
        expect(fn.mock.calls[0][0]).toBe(DATA);

        done();
    });

    it('should have onSubmit method', function () {
        expect(App.prototype.onSubmit).not.toBeUndefined();
    });

    it('should run onNilStartDate function on falsy start date', function (done) {
        const endDate = moment('20171031', 'YYYYMMDD');
        const onNilStartDateMock = jest.fn();
        const div = document.createElement('div');

        ReactDOM.render(<App endDate={endDate} onNilStartDate={onNilStartDateMock}/>, div);
        expect(onNilStartDateMock.mock.calls.length).toBe(1);

        done();
    });

    it('should run onNilEndDate function on falsy start date', function (done) {
        const startDate = moment('20171031', 'YYYYMMDD');
        const onNilEndDateMock = jest.fn();
        const div = document.createElement('div');

        ReactDOM.render(<App startDate={startDate} onNilEndDate={onNilEndDateMock}/>, div);
        expect(onNilEndDateMock.mock.calls.length).toBe(1);
        done();
    });

});