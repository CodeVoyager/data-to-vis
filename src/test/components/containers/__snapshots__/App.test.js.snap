// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<div
  className="app"
>
  <header
    className="header"
  >
    <h1
      className="title"
    >
      Data to Vis
    </h1>
    <h2
      className="sub-title"
    >
      Bitcoin price index chart
    </h2>
  </header>
  <div
    className="steps"
  >
    <div
      className="step"
    >
      <div
        className="header"
      >
        1. Select dates
      </div>
      <div
        className="cols"
      >
        <div
          className="col-half"
        >
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  autoComplete={undefined}
                  autoFocus={undefined}
                  className="input"
                  disabled={false}
                  id={undefined}
                  name={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Click to select start date"
                  readOnly={undefined}
                  required={undefined}
                  tabIndex={undefined}
                  title={undefined}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-half"
        >
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  autoComplete={undefined}
                  autoFocus={undefined}
                  className="input"
                  disabled={false}
                  id={undefined}
                  name={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Click to select end date"
                  readOnly={undefined}
                  required={undefined}
                  tabIndex={undefined}
                  title={undefined}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="step"
    >
      <div
        className="header"
      >
        2. Select currency
      </div>
      <div
        className="cols"
      >
        <div
          className="col-full"
        />
      </div>
    </div>
    <div
      className="step"
    >
      <div
        className="header"
      >
        3. Click submit
        <div
          className="cols"
        >
          <div
            className="col-full"
          >
            <button
              className="input submit"
              disabled={true}
              onClick={[Function]}
            >
              Submit
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App component renders correctly 2`] = `
<div
  className="app"
>
  <header
    className="header"
  >
    <h1
      className="title"
    >
      Data to Vis
    </h1>
    <h2
      className="sub-title"
    >
      Bitcoin price index chart
    </h2>
  </header>
  <div
    className="steps"
  >
    <div
      className="step"
    >
      <div
        className="header"
      >
        1. Select dates
      </div>
      <div
        className="cols"
      >
        <div
          className="col-half"
        >
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  autoComplete={undefined}
                  autoFocus={undefined}
                  className="input"
                  disabled={false}
                  id={undefined}
                  name={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Click to select start date"
                  readOnly={undefined}
                  required={undefined}
                  tabIndex={undefined}
                  title={undefined}
                  type="text"
                  value="2017-10-01"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-half"
        >
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  autoComplete={undefined}
                  autoFocus={undefined}
                  className="input"
                  disabled={false}
                  id={undefined}
                  name={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Click to select end date"
                  readOnly={undefined}
                  required={undefined}
                  tabIndex={undefined}
                  title={undefined}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="step"
    >
      <div
        className="header"
      >
        2. Select currency
      </div>
      <div
        className="cols"
      >
        <div
          className="col-full"
        />
      </div>
    </div>
    <div
      className="step"
    >
      <div
        className="header"
      >
        3. Click submit
        <div
          className="cols"
        >
          <div
            className="col-full"
          >
            <button
              className="input submit"
              disabled={true}
              onClick={[Function]}
            >
              Submit
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App component renders correctly 3`] = `
<div
  className="app"
>
  <header
    className="header"
  >
    <h1
      className="title"
    >
      Data to Vis
    </h1>
    <h2
      className="sub-title"
    >
      Bitcoin price index chart
    </h2>
  </header>
  <div
    className="steps"
  >
    <div
      className="step"
    >
      <div
        className="header"
      >
        1. Select dates
      </div>
      <div
        className="cols"
      >
        <div
          className="col-half"
        >
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  autoComplete={undefined}
                  autoFocus={undefined}
                  className="input"
                  disabled={false}
                  id={undefined}
                  name={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Click to select start date"
                  readOnly={undefined}
                  required={undefined}
                  tabIndex={undefined}
                  title={undefined}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-half"
        >
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  autoComplete={undefined}
                  autoFocus={undefined}
                  className="input"
                  disabled={false}
                  id={undefined}
                  name={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Click to select end date"
                  readOnly={undefined}
                  required={undefined}
                  tabIndex={undefined}
                  title={undefined}
                  type="text"
                  value="2017-10-01"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="step"
    >
      <div
        className="header"
      >
        2. Select currency
      </div>
      <div
        className="cols"
      >
        <div
          className="col-full"
        />
      </div>
    </div>
    <div
      className="step"
    >
      <div
        className="header"
      >
        3. Click submit
        <div
          className="cols"
        >
          <div
            className="col-full"
          >
            <button
              className="input submit"
              disabled={true}
              onClick={[Function]}
            >
              Submit
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App component renders correctly 4`] = `
<div
  className="app"
>
  <header
    className="header"
  >
    <h1
      className="title"
    >
      Data to Vis
    </h1>
    <h2
      className="sub-title"
    >
      Bitcoin price index chart
    </h2>
  </header>
  <div
    className="steps"
  >
    <div
      className="step"
    >
      <div
        className="header"
      >
        1. Select dates
      </div>
      <div
        className="cols"
      >
        <div
          className="col-half"
        >
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  autoComplete={undefined}
                  autoFocus={undefined}
                  className="input"
                  disabled={false}
                  id={undefined}
                  name={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Click to select start date"
                  readOnly={undefined}
                  required={undefined}
                  tabIndex={undefined}
                  title={undefined}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-half"
        >
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  autoComplete={undefined}
                  autoFocus={undefined}
                  className="input"
                  disabled={false}
                  id={undefined}
                  name={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Click to select end date"
                  readOnly={undefined}
                  required={undefined}
                  tabIndex={undefined}
                  title={undefined}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="step"
    >
      <div
        className="header"
      >
        2. Select currency
      </div>
      <div
        className="cols"
      >
        <div
          className="col-full"
        >
          <select
            className="input"
            onChange={[Function]}
            value={undefined}
          >
            <option
              value="USD"
            >
              USD
            </option>
            <option
              value="PLN"
            >
              PLN
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="step"
    >
      <div
        className="header"
      >
        3. Click submit
        <div
          className="cols"
        >
          <div
            className="col-full"
          >
            <button
              className="input submit"
              disabled={false}
              onClick={[Function]}
            >
              Submit
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App component renders correctly 5`] = `
<div
  className="app"
>
  <header
    className="header"
  >
    <h1
      className="title"
    >
      Data to Vis
    </h1>
    <h2
      className="sub-title"
    >
      Bitcoin price index chart
    </h2>
  </header>
  <div
    className="steps"
  >
    <div
      className="step"
    >
      <div
        className="header"
      >
        1. Select dates
      </div>
      <div
        className="cols"
      >
        <div
          className="col-half"
        >
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  autoComplete={undefined}
                  autoFocus={undefined}
                  className="input"
                  disabled={false}
                  id={undefined}
                  name={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Click to select start date"
                  readOnly={undefined}
                  required={undefined}
                  tabIndex={undefined}
                  title={undefined}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-half"
        >
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  autoComplete={undefined}
                  autoFocus={undefined}
                  className="input"
                  disabled={false}
                  id={undefined}
                  name={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Click to select end date"
                  readOnly={undefined}
                  required={undefined}
                  tabIndex={undefined}
                  title={undefined}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="step"
    >
      <div
        className="header"
      >
        2. Select currency
      </div>
      <div
        className="cols"
      >
        <div
          className="col-full"
        />
      </div>
    </div>
    <div
      className="step"
    >
      <div
        className="header"
      >
        3. Click submit
        <div
          className="cols"
        >
          <div
            className="col-full"
          >
            <button
              className="input submit"
              disabled={true}
              onClick={[Function]}
            >
              Submit
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App component renders correctly 6`] = `
<div
  className="app"
>
  <header
    className="header"
  >
    <h1
      className="title"
    >
      Data to Vis
    </h1>
    <h2
      className="sub-title"
    >
      Bitcoin price index chart
    </h2>
  </header>
  <div
    className="steps"
  >
    <div
      className="step"
    >
      <div
        className="header"
      >
        1. Select dates
      </div>
      <div
        className="cols"
      >
        <div
          className="col-half"
        >
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  autoComplete={undefined}
                  autoFocus={undefined}
                  className="input"
                  disabled={false}
                  id={undefined}
                  name={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Click to select start date"
                  readOnly={undefined}
                  required={undefined}
                  tabIndex={undefined}
                  title={undefined}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-half"
        >
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  autoComplete={undefined}
                  autoFocus={undefined}
                  className="input"
                  disabled={false}
                  id={undefined}
                  name={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Click to select end date"
                  readOnly={undefined}
                  required={undefined}
                  tabIndex={undefined}
                  title={undefined}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="step"
    >
      <div
        className="header"
      >
        2. Select currency
      </div>
      <div
        className="cols"
      >
        <div
          className="col-full"
        />
      </div>
    </div>
    <div
      className="step"
    >
      <div
        className="header"
      >
        3. Click submit
        <div
          className="cols"
        >
          <div
            className="col-full"
          >
            <button
              className="input submit"
              disabled={true}
              onClick={[Function]}
            >
              Submit
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="loader-wrapper"
  >
    <div
      className="loader"
    />
  </div>
</div>
`;

exports[`App component renders correctly 7`] = `
<div
  className="app"
>
  <header
    className="header"
  >
    <h1
      className="title"
    >
      Data to Vis
    </h1>
    <h2
      className="sub-title"
    >
      Bitcoin price index chart
    </h2>
  </header>
  <div
    className="steps"
  >
    <div
      className="step"
    >
      <div
        className="header"
      >
        1. Select dates
      </div>
      <div
        className="cols"
      >
        <div
          className="col-half"
        >
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  autoComplete={undefined}
                  autoFocus={undefined}
                  className="input"
                  disabled={false}
                  id={undefined}
                  name={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Click to select start date"
                  readOnly={undefined}
                  required={undefined}
                  tabIndex={undefined}
                  title={undefined}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-half"
        >
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  autoComplete={undefined}
                  autoFocus={undefined}
                  className="input"
                  disabled={false}
                  id={undefined}
                  name={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Click to select end date"
                  readOnly={undefined}
                  required={undefined}
                  tabIndex={undefined}
                  title={undefined}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="step"
    >
      <div
        className="header"
      >
        2. Select currency
      </div>
      <div
        className="cols"
      >
        <div
          className="col-full"
        />
      </div>
    </div>
    <div
      className="step"
    >
      <div
        className="header"
      >
        3. Click submit
        <div
          className="cols"
        >
          <div
            className="col-full"
          >
            <button
              className="input submit"
              disabled={true}
              onClick={[Function]}
            >
              Submit
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App component renders correctly 8`] = `
<div
  className="app"
>
  <header
    className="header"
  >
    <h1
      className="title"
    >
      Data to Vis
    </h1>
    <h2
      className="sub-title"
    >
      Bitcoin price index chart
    </h2>
  </header>
  <div
    className="steps"
  >
    <div
      className="step"
    >
      <div
        className="header"
      >
        1. Select dates
      </div>
      <div
        className="cols"
      >
        <div
          className="col-half"
        >
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  autoComplete={undefined}
                  autoFocus={undefined}
                  className="input"
                  disabled={false}
                  id={undefined}
                  name={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Click to select start date"
                  readOnly={undefined}
                  required={undefined}
                  tabIndex={undefined}
                  title={undefined}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-half"
        >
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  autoComplete={undefined}
                  autoFocus={undefined}
                  className="input"
                  disabled={false}
                  id={undefined}
                  name={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Click to select end date"
                  readOnly={undefined}
                  required={undefined}
                  tabIndex={undefined}
                  title={undefined}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="step"
    >
      <div
        className="header"
      >
        2. Select currency
      </div>
      <div
        className="cols"
      >
        <div
          className="col-full"
        />
      </div>
    </div>
    <div
      className="step"
    >
      <div
        className="header"
      >
        3. Click submit
        <div
          className="cols"
        >
          <div
            className="col-full"
          >
            <button
              className="input submit"
              disabled={true}
              onClick={[Function]}
            >
              Submit
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="loader-wrapper"
  >
    <div
      className="loader"
    />
  </div>
</div>
`;

exports[`App component renders correctly 9`] = `
<div
  className="app"
>
  <header
    className="header"
  >
    <h1
      className="title"
    >
      Data to Vis
    </h1>
    <h2
      className="sub-title"
    >
      Bitcoin price index chart
    </h2>
  </header>
  <div
    className="steps"
  >
    <div
      className="step"
    >
      <div
        className="header"
      >
        1. Select dates
      </div>
      <div
        className="cols"
      >
        <div
          className="col-half"
        >
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  autoComplete={undefined}
                  autoFocus={undefined}
                  className="input"
                  disabled={false}
                  id={undefined}
                  name={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Click to select start date"
                  readOnly={undefined}
                  required={undefined}
                  tabIndex={undefined}
                  title={undefined}
                  type="text"
                  value="2017-10-01"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-half"
        >
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  autoComplete={undefined}
                  autoFocus={undefined}
                  className="input"
                  disabled={false}
                  id={undefined}
                  name={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Click to select end date"
                  readOnly={undefined}
                  required={undefined}
                  tabIndex={undefined}
                  title={undefined}
                  type="text"
                  value="2017-10-01"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="step"
    >
      <div
        className="header"
      >
        2. Select currency
      </div>
      <div
        className="cols"
      >
        <div
          className="col-full"
        >
          <select
            className="input"
            onChange={[Function]}
            value="USD"
          >
            <option
              value="USD"
            >
              USD
            </option>
            <option
              value="PLN"
            >
              PLN
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="step"
    >
      <div
        className="header"
      >
        3. Click submit
        <div
          className="cols"
        >
          <div
            className="col-full"
          >
            <button
              className="input submit"
              disabled={false}
              onClick={[Function]}
            >
              Submit
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="loader-wrapper"
  >
    <div
      className="loader"
    />
  </div>
  <div
    className="loader-wrapper"
  >
    <div
      className="loader"
    />
  </div>
</div>
`;

exports[`App component renders correctly 10`] = `
<div
  className="app"
>
  <header
    className="header"
  >
    <h1
      className="title"
    >
      Data to Vis
    </h1>
    <h2
      className="sub-title"
    >
      Bitcoin price index chart
    </h2>
  </header>
  <div
    className="steps"
  >
    <div
      className="step"
    >
      <div
        className="header"
      >
        1. Select dates
      </div>
      <div
        className="cols"
      >
        <div
          className="col-half"
        >
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  autoComplete={undefined}
                  autoFocus={undefined}
                  className="input"
                  disabled={false}
                  id={undefined}
                  name={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Click to select start date"
                  readOnly={undefined}
                  required={undefined}
                  tabIndex={undefined}
                  title={undefined}
                  type="text"
                  value="2017-10-01"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-half"
        >
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  autoComplete={undefined}
                  autoFocus={undefined}
                  className="input"
                  disabled={false}
                  id={undefined}
                  name={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Click to select end date"
                  readOnly={undefined}
                  required={undefined}
                  tabIndex={undefined}
                  title={undefined}
                  type="text"
                  value="2017-10-01"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="step"
    >
      <div
        className="header"
      >
        2. Select currency
      </div>
      <div
        className="cols"
      >
        <div
          className="col-full"
        >
          <select
            className="input"
            onChange={[Function]}
            value="USD"
          >
            <option
              value="USD"
            >
              USD
            </option>
            <option
              value="PLN"
            >
              PLN
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="step"
    >
      <div
        className="header"
      >
        3. Click submit
        <div
          className="cols"
        >
          <div
            className="col-full"
          >
            <button
              className="input submit"
              disabled={false}
              onClick={[Function]}
            >
              Submit
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="loader-wrapper"
  >
    <div
      className="loader"
    />
  </div>
</div>
`;
